File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minBuy = prices[0], maxSell = 0;
        for(int price : prices) {
            int sellingPrice = price - minBuy;
            maxSell = Math.max(maxSell, sellingPrice);
            minBuy = Math.min(minBuy, price);
        }
        return maxSell;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minBuy = prices[0]
        maxSell = 0
        for price in prices:
            sellingPrice = price - minBuy
            maxSell = max(maxSell, sellingPrice)
            minBuy = min(minBuy, price)
        return maxSell